### ISSUE TITLE
Does not activate if package.json is in a subdirectory

### ISSUE THREAD

**Issue by jasononeil** [xojs/vscode-linter-xo#18]
Hey thanks for the plugin!

I have a project repo which is structured:

```
/
/application <- PHP files
/angular <- JS files
/angular/package.json
```

If I open `./angular/` as the VSC workspace then the linter works correctly.  But if I open the main repo, the linter doesn't seem to activate at all, with no error message given.  Is there any way I can look into what's going on?

If a patch would be needed to support this I'm happy to give it a go!

**Comment by SamVerschueren** [comment]
Thanks for reporting this. You are right that it's not working at the moment. It was reported in the VSCode ESLint repository as well https://github.com/Microsoft/vscode-eslint/issues/32. These projects are quite similar. I would be happy to receive a good PR but I'm not sure if it's that easy to do.

**Comment by wmertens** [comment]
as a workaround, you can symlink your node_modules and package.json in the project root, that makes it work for now but of course clutters your git status.

**Comment by wmertens** [comment]
@SamVerschueren You can use the `pkg-dir` package to find the directory containing the nearest `package.json`, but it seems that your extension assumes a single xo that applies to the whole repo.

I think the extension will need to look up the xo from the correct pkgdir on every call, maybe keeping a cache if that gets slow.

**Comment by oscar-b** [comment]
Any update on this issue? The linked https://github.com/Microsoft/vscode-eslint/issues/32 issue is solved already.

I'm also seeing this error from the plugin:

    [Error - 00:47:00] Request textDocument/codeAction failed.
      Message: Unhandled method textDocument/codeAction
      Code: -32601

**Comment by thiagodebastos** [comment]
Same issue here:
```bash
[Error - 14:41:14] Request textDocument/codeAction failed.
  Message: Unhandled method textDocument/codeAction
  Code: -32601 
```

**Comment by SamVerschueren** [comment]
Tracking that issue here https://github.com/SamVerschueren/vscode-linter-xo/issues/28 now as it is unrelated to this one.

I'm all open for help on this issue as I don't really have much time to look into it.
